src/components/ec-modal/ec-modal.vue

```diff
@@ -0,0 +1,236 @@
+<template>
+  <div
+    v-if="showModal"
+    class="ec-modal"
+  >
+    <div
+      class="ec-modal__content"
+      :class="{'ec-modal--lg': large}"
+    >
+      <header class="ec-modal__header">
+        <slot name="header" />
+        <ec-icon
+          v-if="showCloseIcon"
+          class="ec-modal__close"
+          name="simple-close"
+          :size="24"
+          @click="closeModal()"
+        />
+      </header>
+
+      <main class="ec-modal__main">
+        <slot name="main" />
+      </main>
+
+      <footer class="ec-modal__footer">
+        <div
+          v-if="showHelpLink"
+          v-ec-tooltip="tooltipConfig"
+          class="ec-modal__help-link"
+        >
+          <ec-icon
+            class="ec-mr--8"
+            name="simple-help"
+            :size="18"
+          />
+          Need help?
+        </div>
+
+        <button
+          v-if="this.$slots.negative"
+          class="ec-modal__negative-btn ec-btn ec-btn--md ec-btn--secondary ec-btn--rounded ec-push-right"
+          @click="negativeAction()"
+        >
+          <slot name="negative" />
+        </button>
+
+        <button
+          v-if="this.$slots.positive"
+          :class="{'ec-push-right': !this.$slots.negative}"
+          class="ec-modal__positive-btn ec-btn ec-btn--md ec-btn--primary ec-btn--rounded"
+          @click="positiveAction()"
+        >
+          <slot name="positive" />
+        </button>
+      </footer>
+    </div>
+
+  </div>
+</template>
+
+<script>
+import EcIcon from '../ec-icon';
+import EcTooltip from '../../directives/ec-tooltip/ec-tooltip';
+
+export default {
+  components: {
+    EcIcon,
+  },
+  directives: { EcTooltip },
+  model: {
+    prop: 'showModal',
+    event: 'closeModal',
+  },
+  props: {
+    showModal: {
+      type: Boolean,
+      default: false,
+    },
+    showHelpLink: {
+      type: Boolean,
+      default: false,
+    },
+    showCloseIcon: {
+      type: Boolean,
+      default: true,
+    },
+    large: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      tooltipConfig: {
+        content: "<p>If you are experiencing issues, please send an email to: <a href='mailto:operationsteam@ebury.com'>operationsteam@ebury.com</a></p>",
+        classes: ['ec-tooltip--bg-white'],
+        trigger: 'click',
+        placement: 'bottom',
+      },
+    };
+  },
+  methods: {
+    negativeAction() {
+      this.$emit('negativeAction');
+    },
+    positiveAction() {
+      this.$emit('positiveAction');
+    },
+    closeModal() {
+      this.$emit('closeModal', false);
```

Comment:
I'd remove the word `Modal` from it, because we are going to attach the event handler to modal, we know where we are.

I'd also consider to remove word `Action` from positive and negative events, we are going to type a lot of these. I'd perfectly fine with the code like this:
```
<ec-modal v-model="show" @close="onClose()" @positive="accepted()" @negative="rejected()" />
```